polar-core/src/partial/simplify.rs

```diff
@@ -149,6 +87,84 @@ impl Folder for PartialToExpression {
     }
 }
 
+impl Simplifier {
+    /// Remove duplicate constraints from a partial.
+    fn deduplicate_constraints(&mut self, partial: Partial) -> Partial {
+        let mut seen: HashSet<&Operation> = HashSet::new();
+        let constraints = partial
+            .constraints()
+            .iter()
+            .filter(|o| seen.insert(o))
+            .cloned()
+            .collect();
+        partial.clone_with_constraints(constraints)
+    }
+
+    /// If ``operation`` is an AND or OR operation with 1 argument, return the 1st argument.
+    ///
+    /// Returns: Some(op) if a rewrite occured, or None.
```

Comment:
```suggestion
    /// If `operation` is a 1-arg AND or OR operation, return its argument.
    ///
    /// Returns: Some(op) if a rewrite occurred; otherwise None.
```